### PR TITLE
Feat/tokens transaction history

### LINKED ISSUE #381: Offer management front-end implementation
- [x] Check #256 and fix
- [x] Fix design approval offers by lifebank
- [x] Fix categories in Offers Management by Sponsor

### LINKED ISSUE #705: Backend of tokens history dashboard for lifebank user
Dep: #591

### PR DESCRIPTION
**Pull request by leisterfrancisco**
Resolves #705 

### What does this PR do?
Create the option to a lifebank to see the tokens transaction

### How should this be tested?
- Make run
- Make a lifebank transaction
- Redeem an offer
- Check tokens history in lifebank profile

### CODE REVIEW

File: webapp/src/routes/Home/HomeDesktop.js [hunk 1]
```diff
@@ -86,6 +114,13 @@ const HomeDesktop = (props) => {
             >
               {t('contentToolbar.favorites')}
             </Button>
+            <Button
+              className={classes.buttonIconDesktop}
+              startIcon={<StarIcon />}
+              onClick={redeemOfferHandler}
+            >
+              REDEEM OFFER
+            </Button>
```
Comment (leisterfrancisco) [inline webapp/src/routes/Home/HomeDesktop.js:123]: Only for test purpose

File: webapp/src/routes/Home/HomeDesktop.js [hunk 2]
```diff
@@ -45,6 +52,27 @@ const HomeDesktop = (props) => {
...
+      variables: {
+        ...payload
+      }
+    })
+  }
+
+  useEffect(() => {
+    console.log('REDEEM-OFFER-RESULT', redeemOfferResult)
+  }, [redeemOfferResult])
+
```
Comment (leisterfrancisco) [inline webapp/src/routes/Home/HomeDesktop.js:75]: Only for test purpose

File: webapp/src/routes/Home/HomeDesktop.js [hunk 3]
```diff
@@ -33,6 +35,11 @@ const HomeDesktop = (props) => {
   const [recording, setRecording] = React.useState(false)
   const { transcript } = useSpeechRecognition()
 
+  const [
+    redeemOffer,
+    { loading, error, data: { redeem_offer: redeemOfferResult } = {} }
+  ] = useMutation(REDEEM_OFFER_MUTATION)
+
```
Comment (leisterfrancisco) [inline webapp/src/routes/Home/HomeDesktop.js:42]: Only for test purpose

File: webapp/src/routes/Home/HomeDesktop.js [hunk 4]
```diff
@@ -23,6 +24,7 @@ import MapModal from '../../components/MapModal'
 import FilterHome from '../../components/FilterHome'
 import Signup from '../../components/Signup/Signup'
 import styles from './styles'
+import { REDEEM_OFFER_MUTATION } from '../../gql'
```
Comment (leisterfrancisco) [inline webapp/src/routes/Home/HomeDesktop.js:27]: Only for test purpose

File: hapi/src/api/account.api.js [hunk 5]
```diff
@@ -569,17 +569,26 @@ const transfer = async (from, details) => {
...
+    notification.payload.currentBalance = currentBalance
+    notification.payload.newBalance = newBalance
+    notification.payload.transaction = transaction.transaction_id
+    await notificationApi.insert(notification)
+  } else {
+    await notificationApi.insert({
+      account_from: from,
+      account_to: details.to,
+      title: 'New tokens',
+      description: `From ${from} ${details.memo}`,
```
Comment (JustinCast) [inline hapi/src/api/account.api.js:583]: Shouldn't this be in both languages?

File: hapi/src/api/offer.api.js [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+  if (user.role !== 'donor') {
+    throw new Error('Only donors can redeem an offer')
+  }
+
+  const userTo = await userApi.getOne({
+    account: { _eq: details.to }
+  })
+
+  if (userTo.role !== 'sponsor') {
+    throw new Error('Only sponsor can receive tokens by an offer redemption')
```
Comment (JustinCast) [inline hapi/src/api/offer.api.js:18]: These errors are presented in a snackbar component?

**Comment by JustinCast** [comment]
@leisterfrancisco also please check the conflicts

File: hapi/src/api/offer.api.js (hunk 6 above)
Comment (leisterfrancisco) [reply hapi/src/api/offer.api.js:18]: It is just Backend, front-end integration is in another card

Comment (JustinCast) [reply hapi/src/api/offer.api.js:18]: Got it

File: hapi/src/api/account.api.js (hunk 5 above)
Comment (leisterfrancisco) [reply hapi/src/api/account.api.js:583]: I don't think so, it is for backend logic purpose but I'm not sure at all

**Review by JustinCast — APPROVED** [review]
LGT,

**Review by Edgar097 — APPROVED** [review]

**Review by steph222 — APPROVED** [review]

**Pull request merged**